code/lib/manager-api/src/modules/refs.ts

```diff
@@ -331,9 +331,10 @@ export const init: ModuleFn<SubAPI, SubState> = (
   const initialState: SubState['refs'] = refs;
 
   if (runCheck) {
-    Object.entries(refs).forEach(([id, ref]) => {
-      api.checkRef({ ...ref!, stories: {} } as API_SetRefData);
-    });
+    Object.entries(refs).reduce(async (acc, [id, ref]) => {
+      await acc;
+      await api.checkRef({ ...ref!, stories: {} } as API_SetRefData);
+    }, Promise.resolve());
```

Comment:
This is not the same. 🚨 
Did you test this? the wrapping function isn't async, you can't await here.
![Screenshot 2024-03-08 at 12 02 46](https://github.com/storybookjs/storybook/assets/3070389/9b41128b-0e3a-4fe0-9a7e-13b1aea7c45e)

Would a code comment explaining the reduce help?

Reply:
Ah I see, that makes it even more confusing.

So the `init` function is sync, but internally it runs some async stuff that isn't awaited.

To communicate that I'd think we should create a `new Promise()` that with the async logic inside. That shows that there is async logic happening but it's explicitly not being awaited because the promise isn't awaited anywhere.

```suggestion
    new Promise(async (resolve) => {
      for (const [id, ref] of Object.entries(refs)) {
	      await api.checkRef({ ...ref!, stories: {} } as API_SetRefData);
	    }
	    resolve();
    });
```

If you just want to get on with it, a comment could also work, it just seems like a band aid here to me.